Examples/DotNetCoreBuild/build-with-casaas.sh

```diff
@@ -0,0 +1,212 @@
+    "IsGrpcCopierEnabled ": true
+}
+EOF
+    echo $file
+}
+
+function createCacheConfigJson {
+    local cacheConfigFile=$(mktemp -t cache-config-json)
+    cat > $cacheConfigFile <<EOF
+{
+    "LocalCache": {
+        "CacheId": "SelfhostCS2L1",
+        "Assembly": "BuildXL.Cache.MemoizationStoreAdapter",
+        "CacheLogPath": "[BuildXLSelectedLogPath]",
+        "Type": "BuildXL.Cache.MemoizationStoreAdapter.MemoizationStoreCacheFactory",
+        "CacheRootPath": "${CACHE_ROOT}/cache",
+        "UseStreamCAS": false,
+        "EnableContentServer": true,
+        "EmptyFileHashShortcutEnabled": false,
+        "CheckLocalFiles": false,
+        "CacheName": "${CACHE_NAME}",
+        "GrpcPort": 7089,
+        "Scenario": "${CACHE_SCENARIO}"
+    },
+    "RemoteCache": {
+        "Type": "BuildXL.Cache.BuildCacheAdapter.DistributedBuildCacheFactory",
+        "Assembly": "BuildXL.Cache.BuildCacheAdapter",
+        "CacheId": "RemoteCache",
+        "CacheLogPath": "[DominoSelectedLogPath].L3.log",
+        "CacheServiceFingerprintEndpoint": "https://artifactsu0.artifacts.visualstudio.com/DefaultCollection",
+        "CacheServiceContentEndpoint": "https://artifactsu0.vsblob.visualstudio.com/DefaultCollection",
+        "UseBlobContentHashLists": true,
+        "CacheKeyBumpTimeMins": 120,
+        "CacheNamespace": "${CACHE_NAME}",
+        "CacheName": "${CACHE_NAME}",
+        "ConnectionRetryCount": 5,
+        "ConnectionRetryIntervalSeconds": 5,
+        "GrpcPort": 7089,
+        "SealUnbackedContentHashLists": false,
+        "ConnectionsPerSession": 46,
+        "DisableContent": true
+    },
+    "Type": "BuildXL.Cache.VerticalAggregator.VerticalCacheAggregatorFactory",
+    "Assembly": "BuildXL.Cache.VerticalAggregator",
+    "RemoteContentIsReadOnly": true
+}
+EOF
+    echo $cacheConfigFile
+}
+
+function startContentStoreApp {
+    local settingsFile=$(createContentStoreAppSettingsFile)
+    print_info "Generated ContentStoreApp settings file: ${settingsFile}"
+
+    local casaasArgs=(
+        DistributedService
+        /dataRootPath:${CACHE_ROOT}/dataRootPath
+        /grpcPort:7089
+        /cacheName:${CACHE_NAME}
+        /cachePath:${CACHE_ROOT}/cache
+        /ringId:ring-${CACHE_NAME}
+        /stampId:stamp-${CACHE_NAME}
+        /scenario:${CACHE_SCENARIO}
+        /useDistributedGrpc:true
+        /settingsPath:${settingsFile}
+        /remoteTelemetry
+        /logAutoFlush
+        /logdirectorypath:${CACHE_ROOT}/logs)
+
+    pushd "${CACHE_ROOT}" > /dev/null
+    ${BUILDXL_BIN}/ContentStoreApp "${casaasArgs[@]}" > "${CACHE_STDOUT}" &
+    echo "Exited with code: $?"
+    popd > /dev/null
+}
+
+function runBuildXL {
+    local cacheConfigFile=$(createCacheConfigJson)
+    print_info "Generated BuildXL cache config file: ${cacheConfigFile}"
+
+    pushd "${CACHE_ROOT}" > /dev/null
+    ${MY_DIR}/build.sh --cache-config-file "${cacheConfigFile}" "$@"
+    popd > /dev/null
+}
+
+function validateAndInit {
```

Comment:
>validateAndInit [](start = 9, length = 15)

The following environment variables must also be defined: `CloudStoreRedisConnectionString` and `VSTSPERSONALACCESSTOKEN`

Reply:
Will add those checks as well